fix: allow undefined optional fields in JSON to Borsh serialization

erayack opened the pull request:
## Description

This PR fixes issue #1738 by allowing optional fields to be undefined in JSON input when serializing to Borsh format.

## Problem

Previously, when using the TypeScript SDK, optional fields in call messages (like `relayer` in `TransferRemote`) required explicit `null` values even when they were optional. Omitting these fields would result in the error:

```
failed to call rollup: SerializationError: Expected type or field __SovVirtualWallet_CallMessage_TransferRemote.relayer, but it was not present
```

## Solution

Modified the `visit_struct` method in `json_to_borsh.rs` to:

1. Check if a missing field is an `Option` type
2. If it is optional, create a `Value::Null` instead of throwing an error
3. This allows the `visit_option` method to properly encode the `None` variant

## Changes

- Updated `visit_struct` to handle missing optional fields gracefully
- Added explanatory comment about optional field handling
- Improved error handling for truly missing required fields

## Testing

The fix ensures that TypeScript interfaces with optional fields (marked with `?`) can omit those fields in JSON input, improving the developer experience.

## Related Issue

Fixes #1738

erayack committed:
fix: allow undefined optional fields in JSON to Borsh serialization

- Modified visit_struct to handle missing optional fields by creating Value::Null
- This allows TypeScript SDK to omit optional fields instead of requiring explicit null
- Fixes issue where 'Expected type or field X, but it was not present' error was thrown
- Improves DevX by making optional fields truly optional in JSON input

Resolves #1738

erayack commented:
I have read the CLA Document and I hereby sign the CLA

erayack committed:
remove the comment

preston-evans98 reviewed (approved):
Great change, thanks @erayack! I think this PR is racing with #1931 which makes a nearly identical change - but it looks like yours should merge first. 

Do you have bandwidth to add a test to confirm the new behavior? (Can be done as a follow-up). If not, @theodorebugnet can add one later in the week

erayack commented:
Hi @preston-evans98. I will take a look in an hour to add a test.

erayack committed:
refactor: test rollup creation to support multiple database backends

- Introduced `TestSequencerDbBackend` enum to specify backend type (RocksDb or Postgres).
- Updated `new_test_rollup_with_backend` function to handle Postgres backend initialization.
- Modified existing tests to utilize the new backend support.
- Retained the original `new_test_rollup` function for backward compatibility with RocksDb as the default backend.

erayack commented:
Can you trigger the workflow? @preston-evans98 @theodorebugnet

erayack committed:
refactor: streamline Postgres backend handling in rollup tests

- Updated the `new_test_rollup_with_backend` function to simplify the check for the `SOV_TEST_SKIP_DOCKER` environment variable.
- Moved the `with_postgres_sequencer` function to a more appropriate location within the `RollupBuilder` struct
- Ensured that the Postgres connection setup is properly integrated into the rollup testing framework.

preston-evans98 commented:
The new test looks like it might be for the wrong code. I think all we need to do is call `json_to_borsh` on a hard-coded transaction json string that's missing a `null` field. Shouldn't need any test_rollup changes.

erayack commented:
Thanks, @preston-evans98. I'm a bit confused while addressing the workflow failure.

erayack committed:
- reverted previous tests
- added simple test

erayack committed:
- add serialization test for transaction with missing gas field
